### DISCUSSION TITLE
Image size wrong

### DISCUSSION THREAD

**Discussion by scareduck** [ImageMagick/ImageMagick#4050] [Development] [answered]
Using PerlMagick or the command-line identify tool, I get the wrong dimensions for many photos. The example here comes out as 4608x3456, but in fact is 3456x4608 (portrait not landscape). Firefox gets it right. Ideas?

`[rlm@localhost photos]$ identify img_0068.jpg`
`img_0068.jpg JPEG 4608x3456 4608x3456+0+0 8-bit sRGB 4.66267MiB 0.000u 0:00.002`

![img_0068](https://user-images.githubusercontent.com/30180792/128557900-95060f13-ec97-47e9-b89e-429e8d07acfd.jpg)

**Comment by dlemstra** [comment]
Your image is auto-oriented when being displayed in the browser. The image has a `LeftBottom` orientation instead of  `TopLeft`:

```console
identify -format "%[orientation]" img_0068.jpg
LeftBottom
```

**Comment by fmw42** [comment, marked as answer]
The EXIF data says

```
exif:PixelXDimension: 4608
exif:PixelYDimension: 3456
274 Orientation                     : 8

``` 
which is Orientation: LeftBottom. So the image was captured rotated, but noted where the corner is and should be re-oriented.

Some viewers know about the orientation and will re-orient for you so that it looks portrait, but was actually capture as landscape at 90 deg rotation.

So that is what ImageMagick seems to be telling you.

**Reply by scareduck** [reply]
I had no idea.

I'm trying to get the orientation using the PerlMagick package, but $img->Get('orientation') doesn't seem to work (returns 'Undefined'). 'height' and 'width' it gets correctly, though. I also tried using 'Orientation' (first letter capitalized). What am I doing wrong here?

**Comment by snibgo** [comment]
As the others say. You can get the "portrait" dimensions from:
```
magick img_0068.jpg -auto-orient info:

img_0068.jpg JPEG 3456x4608 3456x4608+0+0 8-bit sRGB 4.66267MiB 0.000u 0:00.001
```

**Reply by scareduck** [reply]
I don't have a magick command-line function. (Fedora 34.)

**Comment by scareduck** [comment]
That. Is. Wild.
Thanks everyone for the assistance!
